hail/python/test/hailtop/utils/test_utils.py

```diff
@@ -126,3 +127,61 @@ def test_parse_docker_image_reference():
     assert x.digest is None
     assert x.name() == 'us-docker.pkg.dev/my-project/my-repo/test-image'
     assert str(x) == 'us-docker.pkg.dev/my-project/my-repo/test-image'
+
+
+
+
+def test_grouped_1():
+    try:
+        actual = list(grouped(0, [1,2,3,4,5,6,7,8,9]))
+        expected = [ 1,2,3,4,5,6,7,8,9]
+    except ValueError:
+        pass
+    else:
+        assert actual == expected
```

Comment:
In plain-English this test says:

- run `grouped(0, ...)`
- if there's a `ValueError` do nothing. (note!!! `pass` has *absolutely nothing to do with tests* it is just the Python expression that means "do nothing")
- if there's no error, check that actual equals expected.

But that's not what we want! *if there is no error*, then the test should fail! We *expect* `grouped(0, [1,2,3,4,5,6,7,8,9])` to fail. It doesn't make sense to make groups of size zero.

Here's an example of a test which verifies that dividing by zero is an error:
```
def test_dividing_by_zero_should_raise_an_error():
    try:
        123 / 0
    except ZeroDivisionError:
        pass  # recall: this just means "do nothing"
    else:
        assert False  # if we get here, then there was *not* an error, so we "assert False" which fails the test
```

I think an important thing to keep in mind is:
- a test fails if the function contains an `assert XXX` where `XXX` evaluates to false.
- otherwise, a test passes.

---

Can you change this test to fail when there is no error?

Reply:
Done